plugins/auth-backend/src/lib/oauth/types.ts

```diff
@@ -78,6 +78,7 @@ export type OAuthState = {
   nonce: string;
   env: string;
   origin?: string;
+  extraState?: string;
```

Comment:
I wonder should this be a hash of strings:

extraState?: { [p: string]: string }

Reply:
I just don't know how this fits with URL encoding though. Will test it out.

Reply:
Can you give me an example of value we expect for extraState?